### PR TITLE
Multi-backups and backup list

### PR DESCRIPTION
**Pull request by itdove**

### CODE REVIEW

File: stories/topics/con-gcp-backup-process.adoc [hunk 1]
```diff
@@ -18,7 +18,9 @@ The backup playbook requires an active {AAPonGCP} foundation deployment to be ru
 
 A bucket needs to be created in your project as restore information will be stored in that bucket.
 
-Only one backup per version is kept in the bucket, if multiple versions of the backup need to be retain then a new bucket must be created for the new version of the backup.
+The bucket can contain multiple backups from the same deployment or different ones. A backup will generate a directory named <prefix>-<deployment_name>-<timestamp> and a file <prefix>-<deployment_name>-<timestamp>.json. The directory contains the awx and pulp databases backup, the deployment configuration and the secrets. The json file contains information for the restore procedure.
```
Comment (ianf77) [inline stories/topics/con-gcp-backup-process.adoc:21]: A backup generates a directory

and a file named <prefix....

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 2]
```diff
@@ -0,0 +1,93 @@
+[id="proc-aws-managing-backups-playbook"]
+
+= Managing backups playbook
+
+There is two playbooks to manage your backups, the first one aws_backup_delete which delete one backup and the second one aws_backups_delete which allows to delete multiple backups at once. We will explain the aws_backups_delete in this chapter. The difference is the aws_backups_delete takes a array of string ["backup1","backup2",...] but the aws_backup_delete take only one string which is the name of a specifc backup "backup1"
+
+== Delete backups
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:7]: This should be a separate procedure.
Deleting backups

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+----
+TASK [redhat.ansible_on_clouds.standalone_aws_backup_delete : [delete_backup] Dry-run message] ***
+skipping: [localhost]
+
+PLAY RECAP *********************************************************************
+localhost                  : ok=21   changed=2    unreachable=0    failed=0    skipped=2    rescued=0    ignored=0    
+----
+
+
+= Troubleshootig
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:79]: Troubleshooting

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+== Failed to delete a backup
+
+If the deletion of backup failed, take the following steps.
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:83]: fails,

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+1. Go to the bucket containing the backup and locate the directory having the backup name.
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:85]: You don't need to number these steps, asciidoc numbers them automatically.

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+. Populate the `command_generator_data` directory with the configuration file template.
++
+[source,bash]
+----
+docker run --rm -v $(pwd)/command_generator_data/:/data $IMAGE command_generator_vars aws_backups_delete --output-data-file /data/backups.yml
+----
++
+Produces the following output:
++
+[literal, options="nowrap" subs="+quotes,attributes"]
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:18]: remove +quotes, from the header

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+-----------------------------------------------
+Command generator template: 
+
+docker run --rm -v <local_data_file_directory>:/data $IMAGE command_generator aws_backups_delete --data-file /data/backups.yml
+----
+. After running the command, a `/tmp/backups.yml` template file is created. 
+This template file resembles the following: 
++
+[literal, options="nowrap" subs="+quotes,attributes"]
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:34]: remove +quotes, from the header

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    aws_region:
+    aws_s3_bucket:
+    delete:
+----
+
+With backup_names like ["backup1","backup2"]
+
+. To run the deletion of the backups, run the command generator to generate the deletion backup command.
++
+[literal, options="nowrap" subs="+quotes,attributes"]
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:49]: remove +quotes,

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+----
+docker run --rm -v $(pwd)/command_generator_data:/data $IMAGE command_generator aws_backups_delete --data-file /data/backups.yml
+----
++
+Resulting in the following ouput:
++
+[literal, options="nowrap" subs="+quotes,attributes"]
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:56]: remove +quotes,

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+docker run --rm --env PLATFORM=AWS -v ~/.aws/credentials:/home/runner/.aws/credentials:ro \
+--env ANSIBLE_CONFIG=../aws-ansible.cfg  $IMAGE redhat.ansible_on_clouds.aws_backups_delete \
+-e 'aws_region=<region> aws_s3_bucket=<bucket> aws_backup_names=["mybackup"] delete=True'
+===============================================
+----
+. Run the supplied backup command to trigger the backup deletion.
++
+. When the playbook has finished running, the output resembles the following:
++
+[literal, options="nowrap" subs="+quotes,attributes"]
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:69]: remove +quotes,

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+
+= Troubleshootig
+
+== Failed to delete a backup
+
+If the deletion of backup failed, take the following steps.
+
+1. Go to the bucket containing the backup and locate the directory having the backup name.
+2. Open that directory
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:86]: . Open the backup directory.

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+3. Open the file names "restore-vars.json" and note last part of the "Recovery_Point_Arn" and the "Backup_Vault_Name"
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:87]: `restore-vars.json ` is one file. Does it contain both peices of information.

File: stories/topics/proc-aws-managing-backups-playbook.adoc [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+4. On the AWS Console, go to AWS Backup
```
Comment (ianf77) [inline stories/topics/proc-aws-managing-backups-playbook.adoc:88]: navigate to * AWS backup *.

**Review by ianf77 — COMMENTED** [review]
Some comments. More to follow.

File: stories/topics/con-gcp-backup-process.adoc (hunk 1 above)
Comment (itdove) [reply stories/topics/con-gcp-backup-process.adoc:21]: The file is postfix by json and doesn't so doesn't have the same name as the directory.

Comment (itdove) [reply stories/topics/con-gcp-backup-process.adoc:21]: ok got it you want me to add the workd "named" after the "file"

File: stories/topics/proc-aws-managing-backups-playbook.adoc (hunk 2 above)
Comment (itdove) [reply stories/topics/proc-aws-managing-backups-playbook.adoc:7]: I renamed the proc as there is only 1 procedure in that document.

File: stories/topics/proc-aws-managing-backups-playbook.adoc (hunk 3 above)
Comment (itdove) [reply stories/topics/proc-aws-managing-backups-playbook.adoc:79]: Created a new proc file for the troubleshooting

File: stories/topics/proc-aws-managing-backups-playbook.adoc (hunk 4 above)
Comment (itdove) [reply stories/topics/proc-aws-managing-backups-playbook.adoc:83]: fixed

File: stories/topics/proc-aws-managing-backups-playbook.adoc (hunk 6 above)
Comment (itdove) [reply stories/topics/proc-aws-managing-backups-playbook.adoc:18]: Done

File: stories/topics/proc-aws-managing-backups-playbook.adoc (hunk 7 above)
Comment (itdove) [reply stories/topics/proc-aws-managing-backups-playbook.adoc:34]: Done

File: stories/topics/proc-aws-managing-backups-playbook.adoc (hunk 9 above)
Comment (itdove) [reply stories/topics/proc-aws-managing-backups-playbook.adoc:56]: Done

File: stories/topics/proc-aws-managing-backups-playbook.adoc (hunk 10 above)
Comment (itdove) [reply stories/topics/proc-aws-managing-backups-playbook.adoc:69]: Done

File: stories/topics/proc-aws-managing-backups-playbook.adoc (hunk 11 above)
Comment (itdove) [reply stories/topics/proc-aws-managing-backups-playbook.adoc:86]: Done

File: stories/topics/proc-aws-managing-backups-playbook.adoc (hunk 12 above)
Comment (itdove) [reply stories/topics/proc-aws-managing-backups-playbook.adoc:87]: yes

File: stories/topics/proc-aws-managing-backups-playbook.adoc (hunk 13 above)
Comment (itdove) [reply stories/topics/proc-aws-managing-backups-playbook.adoc:88]: Done

**Review by ianf77 — APPROVED** [review]
LGTM

**Pull request merged**